app/uk/gov/hmrc/helptosavefrontend/connectors/NSAndIConnector.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.connectors
+
+import java.time.LocalDate
+
+import com.google.inject.ImplementedBy
+import uk.gov.hmrc.helptosavefrontend.WSHttp
+import uk.gov.hmrc.helptosavefrontend.models.NSIUserInfo
+import uk.gov.hmrc.helptosavefrontend.util.Result
+import uk.gov.hmrc.play.config.ServicesConfig
+import uk.gov.hmrc.play.http._
+import javax.inject.Singleton
+
+import play.api.Logger
+import play.api.http.Status
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+case class SubmissionSuccess()
+
+case class SubmissionError(code: Int, message: String)
+
+@ImplementedBy(classOf[NSAndIConnectorImpl])
+trait NSAndIConnector {
+  type NSAndIResponse = Either[SubmissionError, SubmissionSuccess]
+
+  def createAccount(userINfo: NSIUserInfo)(implicit hc: HeaderCarrier): Future[NSAndIResponse]
+
+  val nSAndIUrl: String
+  val http: HttpGet with HttpPost
```

Comment:
Our other connectors have these variables in the implementation rather than the trait